tyk-hybrid/templates/deployment-gw-repset.yaml

```diff
@@ -92,11 +89,25 @@ spec:
           - name: TYK_GW_SLAVEOPTIONS_BINDTOSLUGSINSTEADOFLISTENPATHS
             value: "{{ default "false" .Values.gateway.rpc.bindToSlugs }}"
           - name: TYK_GW_SLAVEOPTIONS_SSLINSECURESKIPVERIFY
-            value: "{{ .Values.gateway.rpc.sslInsecureSkipVerify }}"  
+            value: "{{ .Values.gateway.rpc.sslInsecureSkipVerify }}"
+        # Enable backward compatibility to chart 0.5.1.
+        {{- if or .Values.gateway.sharding .Values.enableSharding}}
           - name: TYK_GW_DBAPPCONFOPTIONS_NODEISSEGMENTED
+          {{- if .Values.gateway.sharding }}
+            value: "{{ .Values.gateway.sharding.enabled }}"
+          {{- else }}
             value: "{{ .Values.enableSharding }}"
+          {{- end }}
+        {{- end }}
+        # Enable backward compatibility to chart 0.5.1.
+        {{- if or .Values.gateway.sharding .Values.enableSharding}}
           - name: TYK_GW_DBAPPCONFOPTIONS_TAGS
-            value: "{{.Values.gateway.tags }}"
+          {{- if .Values.gateway.sharding }}
```

Comment:
I guess the condition should check whether `.Values.gateway.tags` is empty or not and based on that it'd update the value.

Reply:
Resolving this in favour of the comment above.